src/@chakra-ui/gatsby-plugin/styles.ts

```diff
@@ -17,16 +15,18 @@ const styles = {
       // TODO: when we have Chakra v2, this should be done by overriding the
       // native Chakra semantic tokens
       bg: mode("white", "gray.700")(props),
-      lineHeight: "1.6rem",
+      lineHeight: "base",
+      fontSize: ["sm", null, "lg"],
+    },
+    p: {
+      _last: {
+        mb: 0,
+      },
     },
     a: {
       color: "primary.base",
       textDecoration: "underline",
     },
-    // should be replace with https://chakra-ui.com/docs/components/text
-    p: {
-      margin: "0px 0px 1.45rem",
```

Comment:
Now, looking at the preview deploy, I see that we kind of depend on this margin to avoid having to refactor all the codebase. I'd say that for the `Text` component we add this button margin. What do you think @nloureiro and @TylerAPfledderer?

Note for context for @nloureiro. This `Text` component represents the `p` tag, the paragraphs in html.

Reply:
@pettinarip fair point!

So for example, with the Homepage hero where this margin would be used against the CTA button "Explore Ethereum", when working on the new heroes, I took a different approach, whereby I structured the Header and text under one flex stack, and then that grouping with the button at a higher level stack, and used the `gap` or `spacing` prop to define the "margin" between the different texts and between the paragraph and the button.

I should ask then what would best practice be here? In my description above, this approach using the Flexbox `gap` property to define the margins between content instead of using the `margin` prop.

And I would consider in addition to the above the idea of using `margin` only when we are stacking paragraphs, with a defined :first- or :last-of-type selector so that only margin is defined between the paragraphs, but not added above the first or below the last of the stacked siblings.

So:
```ts
"p:not(:last-of-type)": {
  mb: "2" // or some other token value
}
```

I also understand that margin has to be well defined for the mdx file parsing because you have no ability to isolate groups of text elements into neatly packaged boxes. So I would want to figure out what the best practice should be to keep usage in MDX pages inline with the page layouts that do not use MDX.

Reply:
Using `gap` and `spacing` is correct and expected as the best way to go looking ahead.

Now, talking about the current problem and thinking about how to support the old margins, I think that the best way to go is by leaving the `Text` component as it is and creating a new temporary one called `OldText` that would have the old margins.

Following this idea
- the `Text` component would be used in all new implementation
- the `OldText` would be marked as deprecated and would be used in the rest of the old/current implementations.

The `OldText` component might be something as simple as this
```tsx
function OldText(props) {
  return <Text margin={oldDefaultMargins} {...props} />
}
```

With that new component in place, I could go and refactor the entire codebase by importing the `OldText` and adding those changes to this PR.

Then, as you migrate the new components as the `Hero`, you can start using and importing the native `Text`.

Does it make sense?

Reply:
@pettinarip Ok! So using the `OldText` component for the MDX files too, correct?

Reply:
Correct.

Reply:
@pettinarip I've pushed this new component here.

Reply:
Thanks. I'll refactor the code to use that component instead of the `Text`.

Reply:
Sounds good!